File: Simulacao/MM2_Python/GNA.py
import random
from datetime import datetime

class Seed:
    def __init__(self, InitSeed = datetime.now().microsecond):
        self.seed = (2 * InitSeed) - 1  #GARANTIA DE SEED SER IMPAR
    
    def getSeed(self):
        return self.seed

    def setSeed(self, newSeed):
        self.seed = newSeed

def MCLM(seed):
    #CONSTANTES
    a = 19           #MULTIPLICADOR
    m = 128          #MODULO
    seed.setSeed(((a * seed.getSeed()) % m))
    return seed.getSeed()/float(m)

File: Simulacao/MM_1/GNA.js

class Seed{
    
    constructor(InitSeed = new Date()){
        this.seed = (2*InitSeed) - 1;
    }

    get Seed(){
        return this.seed;
    }

    set Seed(newSeed){
        this.seed = newSeed;
    }
}
    

function MCLM(seedObj){
    const a = 19;
    const m = 128;
    seedObj.seed = (((a * seedObj.seed % m)));
    return seedObj.seed/m;
}
